### DISCUSSION TITLE
Indexing issues - GPS & Timezone

### DISCUSSION THREAD

**Discussion by DustyTools** [photoprism/photoprism#1367] [General]
Hi All,

So I am new to PhotoPrism and playing around with the import. I copied over some test files and ran the index command. I got a bunch of errors that the GPS data couldn't be read, and when I look at the photo in the UX it shows them in UTC time.

How do I go about debugging the issues?

All the GPS was written using a library I wrote years ago (https://github.com/DustyTools/fotofly), I am confident the data is written correctly. I did extensive testing using a few different apps...

I am also unclear how the timezone is detected? When I was writing that library there was no standard way to handle timezones. I wrote the data in a proprietary IPTC property, is there now a standard way to store this data? I can migrate it over....

Thanks

Ben

**Comment by DustyTools** [comment]
Here's an example of what I am seeing

![Capture](https://user-images.githubusercontent.com/2087028/121727587-f3df2e00-caa0-11eb-982c-9d2a20e4a65f.PNG)

**Reply by lastzero** [reply]
Seems to be stored in a non standard / not common format. What is it?

**Reply by DustyTools** [reply]
The coordinates appear to be valid...

![Capture](https://user-images.githubusercontent.com/2087028/121737918-5ee33180-caae-11eb-9148-84c6611f56d9.PNG)

They are written using my own software, I did extensive testing and validation when I wrote it many years ago. This is the first time in ages that the data can't be read.

**Reply by lastzero** [reply]
Should be a float number. Where is this from?

**Reply by DustyTools** [reply]
Oh that's a screen shot of:

https://jimpl.com/results/jRtUrb5pvVSTErzvcBPwFQ6V

Not sure if that will load for you.... It loads for me in another browser, which is a disturbing privacy issue!

**Reply by lastzero** [reply]
We can test it after our vacation. Not in office currently.

**Reply by DustyTools** [reply]
No problem! I can fix the data if it's not stored correctly, I'm just not sure what's wrong.

**Reply by lastzero** [reply]
Did you have time to look at this? I recommend comparing working files with GPS coordinates (e.g. from your smartphone) with your edited files to see if there are any differences in the storage format or if certain files are missing (there are multiple GPS* fields in Exif).

**Comment by DustyTools** [comment]
Sorry for the delay. I tested today and I can't see how the data I'm writing is incorrect.

Here's two screenshots from a tool called WICExplorer, it just shows the raw data.

In this screen shot you can see data written by a iPhone X, it's populated the GPS created data, which is the four values #1-4. #2 & #4 store the numeric values which are written as three urationals:

![Git Hub 1](https://user-images.githubusercontent.com/2087028/129502122-b15ba477-4c60-46b1-8277-9c5a4c19156e.JPG)

In this screenshot you can see the same data is preserved. But #22 and #24 (GPS shown data) are now set, also with three urationals. 

![Github 2](https://user-images.githubusercontent.com/2087028/129502123-b5432eb0-a1e0-4982-b148-7c8173f6cbac.JPG)

Admittedly the format is a little different, but they are valid. So maybe PhotoPrism is expecting a specific format for the denominator? I checked this file using a few online viewers and they all pick up the data correctly.

My guess is in gps.go, func GpsToLatLng(s string) (lat, lng float32) Minutes is expected to be a whole number but in my files it's a fraction.

I don't remember why I formatted it this way. My code has an override that allows the seconds to be a decimal of the minutes and that's how I've been doing it for 10 years...

So the three URationals can either be stored as:

{ 47 / 1 , 39 / 1 , 235 / 100  }

or

{ 47 / 1 , 39392 / 1000 (39.392), 0 / 0 } <- Where 0.392 is 23.5/60 

I'm writing a parser to touch my files and format them the same way as an iPhone, so I probably won't hit this bug again, but it might effect others.

Ben

**Reply by lastzero** [reply]
Sadly we have an overflowing todo list, and not much time to experiment at the moment. Extracting GPS coordinates works in general with images tagged by common cameras and mobile phones. When I have a few free minutes, it would be helpful to have some example files to test with. Feel free to send them to hello@photoprism.app!
